php/class-photonfill.php

```diff
@@ -378,6 +382,26 @@ public function image_size_names_choose( $data ) {
 		}
 
 		/**
+		 * Add image caption requires element to have a width to display shortcode correctly
+		 */
+		public function add_width_for_captions( $html, $id, $caption, $title, $align, $url, $size, $alt = '' ) {
+			$caption = apply_filters( 'image_add_caption_text', $caption, $id );
+			if ( ! empty( $caption ) ) {
+				if ( is_numeric( $size ) ) {
+					$size_px = $size;
+				} elseif ( is_array( $size ) ) {
+					$size_px = $size[0];
+				} else {
```

Comment:
If size is a string like `'medium'` it might be worth looking up that image size and setting the width based on that before getting the original size of the attachment with `wp_get_attachment_metadata()`